### PR TITLE
[DEVOPS-1032] nix: Add flag to build with optimization disabled

### PR DESCRIPTION
**Pull request by rvl**
## Description

Adds a `fasterBuild` argument to the nix build. This will result in `-O0` being passed to GHC. This can be enabled for PR builds (see input-output-hk/iohk-ops#418) to get quicker turnaround.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/DEVOPS-1032

### CODE REVIEW

**Comment by rvl** [comment]
Rebased on latest develop with temp testing commits removed.

File: default.nix [hunk 1]
```diff
@@ -111,6 +112,14 @@ let
     });
   };
 
+  # Disabling optimization for cardano-sl packages will
+  # return a build x% faster (measured in DEVOPS-1032).
```
Comment (jmitchell) [inline default.nix]: According to the issue both mine and Rodney's experiments outside Hydra with cache invalidation (`echo >> util/cardano-sl-util.cabal`) shows ~20% speedup for `all-cardano-sl` when optimization is disabled.

On that basis, let's `s/x%/20%/` here. https://github.com/input-output-hk/iohk-ops/pull/418 takes care of passing `fasterBuild = true;` in the Hydra jobset only for PRs.

Comment (rvl) [reply default.nix]: OK yes I have amended the commit to add that figure.

**Comment by rvl** [comment]
@jmitchell I realised that I forgot to pass through the argument in `release.nix` (required for input-output-hk/iohk-ops#418).

I have pushed another commit and tested `nix-build release.nix --arg fasterBuild true -A cardano-sl.x86_64-linux`.

**Pull request merged**